index.js

```diff
@@ -80,44 +84,117 @@ if (result.error) {
     throw result.error;
 }
 
-// Instantiate bomo
-const bomo = new Bomo();
+const initialize = async function() {
+    // ES Modules in CommonJS via dynamic import
+    const { rateLimit } = await import("@tinyhttp/rate-limit");
 
-// Setting engine, logging middleware, 404 route, and serving the public folder are handled by Bomo's constructor
+    // Instantiate bomo
+    const bomo = new Bomo();
 
-// Register api routes with tinyhttp
-bomo.app.get("/time", (req, res, next) => res.json({ content: DateTime.now().toFormat("HH:mm:ss.SSS") }));
-bomo.app.post("/time", (req, res, next) => res.json({
-    message: "yes i can here u are",
-    content: DateTime.now().toFormat("HH:mm:ss.SSS"),
-}));
+    // Setting engine, parsing cookie headers, logging middleware, 404 route, and serving the public folder are handled by Bomo's constructor
 
-// Lobby API Endpoints
-bomo.app.post("/api/lobby/create", (req, res, next) => {
-    // params: name, some sort of browser/session id
-    // returns: success, lobby code
-});
+    // Register api routes with tinyhttp
+    // bomo.app.get("/time", (req, res, next) => res.json({ content: DateTime.now().toFormat("HH:mm:ss.SSS") }));
+    // bomo.app.post("/time", (req, res, next) => res.json({
+    //     message: "yes i can here u are",
+    //     content: DateTime.now().toFormat("HH:mm:ss.SSS"),
+    // }));
 
-bomo.app.post("/api/lobby/join", (rep, res, next) => {
-    // params: name, some sort of browser/session id, lobby code
-    // returns: success
-});
+    console.debug(bomo.auth.generate);
+
+    /**
+     * Creates a new session for authentication
+     */
+    bomo.app.post("/api/sessions",
+        rateLimit({ max: 16, windowMs: 86400000 /* 24 hours */ }),
+        /** @todo For whatever reason, using the generate method as a handler directly causes a 500 internal server error, with nothing apparently wrong. ive spent the past several hours debugging it to no avail, so you have the solution you see before you, which just works? why?? have i ever?? */
```

Comment:
This issue was baffling to me. I couldn't narrow it down to it being that the method was async, an issue with using `this`, etc. It would just always return 500 internal server error. Which is an issue for debugging, because I still have no idea what causes it.

Reply:
I figured out what was causing this btw. It has to do with tinyhttp's options and iirc i fixed it